client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/TrainedModelConfig.java

```diff
@@ -226,6 +235,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (licenseLevel != null) {
             builder.field(LICENSE_LEVEL.getPreferredName(), licenseLevel);
         }
+        if (defaultFieldMappings != null) {
```

Comment:
```suggestion
        if (defaultFieldMappings != null && defaultFieldMappings.isEmpty() == false) {
```

I prefer not to write empty collections.